osquery/remote/enrollment/plugins/remote_enrollment.cpp

```diff
@@ -0,0 +1,86 @@
+#include <osquery/enrollment.h>
+
+#include "osquery/remote/requests.h"
+#include "osquery/remote/transports/http.h"
+#include "osquery/remote/serializers/json.h"
+#include <boost/property_tree/json_parser.hpp>
+
+#define MAX_TRIES 5
+
+namespace osquery {
+
+FLAG(string,
+     enrollment_uri,
+     "Not Specified",
+     "The endpoint for server side client enrollment");
+
+FLAG(string,
+     enrollment_secret,
```

Comment:
Also, food for thought. Given that we don't have a good way to let plugins yield additional flags. We should be careful to design our included enrollment plugins using the language and flexibility of a very-basic-TLS-enrollment process. Adding variations that subclass a usable TLS-enrollment plugin is possible too.